tests/test_file.py

```diff
@@ -414,6 +421,67 @@ def try_get():
                 url, requests.codes.bad_request)
         try_get()
 
+    @testmode.integration
+    def test_file_get_checkout(self):
+        """
+        Verifies checkout occurs on first get and not on second.
+        :return:
+        """
+        tempdir = tempfile.gettempdir()
+        self._test_file_get_checkout(Replica.aws, "s3", self.s3_test_bucket,
+                                     S3Uploader(tempdir, self.s3_test_bucket))
+        self._test_file_get_checkout(Replica.gcp, "gs", self.gs_test_bucket,
+                                     GSUploader(tempdir, self.gs_test_bucket))
+
+    def _test_file_get_checkout(self, replica: Replica, scheme: str, test_bucket: str, uploader: Uploader):
+        src_key = generate_test_key()
+        src_data = os.urandom(1024)
+        with tempfile.NamedTemporaryFile(delete=True) as fh:
+            fh.write(src_data)
+            fh.flush()
+
+            uploader.checksum_and_upload_file(fh.name, src_key, "text/plain")
+
+        source_url = f"{scheme}://{test_bucket}/{src_key}"
+
+        file_uuid = str(uuid.uuid4())
+        bundle_uuid = str(uuid.uuid4())
+        version = datetime_to_version_format(datetime.datetime.utcnow())
+
+        # should be able to do this twice (i.e., same payload, different UUIDs)
+        self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid, version=version)
+        url = str(UrlBuilder()
+                  .set(path="/v1/files/" + file_uuid)
+                  .add_query("replica", replica.name)
+                  .add_query("version", version))
+
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        @eventually(20, 2)
+        def test_checkout():
+            # assert 302 found and checksum when checkout completes successfully
+            api_get = self.assertGetResponse(
+                url, requests.codes.found, redirect_follow_retries=0)
+            file_get = requests.get(api_get.response.headers['Location'])
+            self.assertTrue(file_get.ok)
+            self.assertEquals(file_get.content, src_data)
+
+        # assert 301 redirect on first GET
+        self.assertGetResponse(url, requests.codes.moved, redirect_follow_retries=0)
+        test_checkout()
+
+        last_modified_fn = ("cloud_blobstore.s3.S3BlobStore.get_last_modified_date"
+                            if replica.name == "aws"
+                            else "cloud_blobstore.gs.GSBlobStore.get_last_modified_date")
+        with mock.patch(last_modified_fn) as mock_last_modified:
+            # assert 301 redirect on stale file
+            blob_ttl_days = int(os.environ['DSS_BLOB_PUBLIC_TTL_DAYS'])
```

Comment:
rather than mock the `get_last_modified date` call, you could make the TTL a second-based variable, override that to some very low value, and then do your second get (which should be "stale")

Reply:
In other words, modify `DSS_BLOB_PUBLIC_TTL_DAYS` to be expressed in seconds such that it can be mocked in terms of seconds?

Reply:
For ease, you may still want the environment variable to be in units of days, but when read in, you could store it in seconds, and then tweak your behavior based on that.

Reply:
Sounds like another layer of complexity that would exist to modify the implementation of the test. I'd argue to keep this as it is